swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
+from swap_meet.item import Item
 class Vendor:
-    pass
+    def __init__(self, inventory=[]):
+        self.inventory = inventory 
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if len(self.inventory) > 0:
+            for thing in self.inventory:
+                if thing == item:
+                    self.inventory.remove(item)
+                    return item
+        return False
+
+    def get_by_category(self,category):
+        category_list = []
+        for thing in self.inventory:
+            if thing.category == category:
+                category_list.append(thing)
+        return category_list
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.remove(my_item)
+            friend.remove(their_item)
+            friend.add(my_item)
+            self.add(their_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, friend):
+        if len(self.inventory) == 0 or len(friend.inventory) == 0:
+            return False
+        else:
+            my_item = self.inventory.pop(0)
+            friend.add(my_item)
+            their_item = friend.inventory.pop(0)
+            self.add(their_item)
+            return True
+
+    def get_best_by_category(self, category):
+        highest_item = None
+        for item in self.inventory:
```

Comment:
Could we reuse another method to get all the items of a particular category? How would that affect the readability of the code here? What if we used Python built-in functionality to find the max? Is there a difference in the complexity of getting the filtered list first and then finding the max than there is with filtering and finding at the same time?